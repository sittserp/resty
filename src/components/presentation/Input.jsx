import React from 'react';
import PropTypes from 'prop-types';

const Input = ({ url, method, body, onSubmit, onChange }) => {
  return (
    <>
      <form onSubmit={onSubmit}>
        <input 
          type="text" 
          name="url" 
          placeholder="URL" 
          value={url} 
          onChange={onChange} />

        <div>
          <input 
            type="radio" 
            id="get" 
            name="method" 
            value="GET" 
            checked={method === 'GET'} 
            onChange={onChange} />
          <label htmlFor="get">GET</label>

          <input 
            type="radio" 
            id="post" 
            name="method" 
            value="POST" 
            checked={method === 'POST'} 
            onChange={onChange} />
          <label htmlFor="post">POST</label>
            
          <input 
            type="radio" 
            id="put" 
            name="method" 
            value="PUT" 
            checked={method === 'PUT'} 
            onChange={onChange} />
          <label htmlFor="put">PUT</label> 
            
          <input 
            type="radio" 
            id="patch" 
            name="method" 
            value="PATCH" 
            checked={method === 'PATCH'} 
            onChange={onChange} /> 
          <label htmlFor="patch">PATCH</label>
            
          <input 
            type="radio" 
            id="delete" 
            name="method" 
            value="DELETE" 
            checked={method === 'DELETE'} 
            onChange={onChange} />
          <label htmlFor="delete">DELETE</label>
        </div>
        <button>Call</button>
        <textarea 
          placeholder="JSON Body" 
          name="body" 
          value={body} 
          onChange={onChange}></textarea>
      </form>
    </>
  );
};

Input.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Input;
