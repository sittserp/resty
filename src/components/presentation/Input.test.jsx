import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Input from './Input';

describe('Input component', () => {
  afterEach(() => cleanup());
  it('renders Input', () => {
    const { asFragment } = render(<Input 
      url="dummy string"
      method="dummy string"
      body="dummy string"
      onChange={jest.fn()}
      onSubmit={jest.fn()}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
