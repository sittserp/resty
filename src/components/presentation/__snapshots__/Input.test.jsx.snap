// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders Input 1`] = `
<DocumentFragment>
  <form>
    <input
      name="url"
      placeholder="URL"
      type="text"
      value="dummy string"
    />
    <div>
      <input
        id="get"
        name="method"
        type="radio"
        value="GET"
      />
      <label
        for="get"
      >
        GET
      </label>
      <input
        id="post"
        name="method"
        type="radio"
        value="POST"
      />
      <label
        for="post"
      >
        POST
      </label>
      <input
        id="put"
        name="method"
        type="radio"
        value="PUT"
      />
      <label
        for="put"
      >
        PUT
      </label>
      <input
        id="patch"
        name="method"
        type="radio"
        value="PATCH"
      />
      <label
        for="patch"
      >
        PATCH
      </label>
      <input
        id="delete"
        name="method"
        type="radio"
        value="DELETE"
      />
      <label
        for="delete"
      >
        DELETE
      </label>
    </div>
    <button>
      Call
    </button>
    <textarea
      name="body"
      placeholder="JSON Body"
    >
      dummy string
    </textarea>
  </form>
</DocumentFragment>
`;
